### DISCUSSION TITLE
Multiple links between elements?

### DISCUSSION THREAD

**Discussion by dmb0058** [clientIO/joint#2547] [Q&A] [answered]
### Introduction

Trying to get https://resources.jointjs.com/tutorial/multiple-links-between-elements working using JointJS 4.0.1 but I get an error "Uncaught ReferenceError: _ is not defined".

This occurs in multiple places, e.g.

```
var adjustGraphVertices = _.partial(adjustVertices, graph);
```

I'm pretty sure I've used this approach successfully in JointJS 3.7.7, is the graphlib split out as a separate component or something?

David

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
The example is using [lodash](https://lodash.com/) (not `graphlib`). The `v4` version has no dependencies, so you have to install it separately.

**Comment by dmb0058** [comment]
Aha! yes, that makes sense. I'll see if I can find a v4 version ... a link
would be amazing if there is a version for v4 😊

Thanks,

David

**Reply by kumilingus** [reply]
I was referring to _JointJS_ `v4`.  Based on your question, you are already using it.

> Trying to get https://resources.jointjs.com/tutorial/multiple-links-between-elements working using JointJS 4.0.1 but I get an error "Uncaught ReferenceError: _ is not defined".

**Reply by dmb0058** [reply]
Yes, I was too ... I'm using JointJS 4.0.1.

What I meant was "I'll see if I can find a JointJS 4.0 compatible version of the 'Multiple links between elements' code at the link above".

But I realised that you meant "so you should just install lodash and use that", and you're right, it works fine.

Thanks,

David
